include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -581,14 +753,28 @@ __pattern_copy_if(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _Range1&
 }
 
 #if _ONEDPL_CPP20_RANGES_PRESENT
+
+template <typename _Pred, typename _Proj>
+struct __pattern_copy_if_ranges_pred
```

Comment:
We already defined a type with the same functionality above. And we can re-use one.